### PR TITLE
Move messages code to right subdir

### PR DESCRIPTION
**Pull request by michael-loki**

### CODE REVIEW

**Comment by majestrate** [comment]
this may conflict with @tewinget 's LRSM changes

**Comment by michael-loki** [comment]
Since it’s mostly a clean move, I think a merge would probably resolve it?

**Comment by majestrate** [comment]
On Mon, Jun 17, 2019 at 05:37:34AM -0700, michael-loki wrote:
> Since it’s mostly a clean move, I think a merge would probably resolve it?

i'd like to verify with tom first just to be sure.

File: llarp/messages/link_message_parser.cpp [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+#include <router_contact.hpp>
+#include <util/buffer.hpp>
+#include <util/logger.hpp>
+#include <util/metrics.hpp>
+
+#include <memory>
+
+namespace llarp
+{
+  struct InboundMessageParser::msg_holder_t
```
Comment (tewinget) [inline llarp/messages/link_message_parser.cpp]: given the nature of this PR, perhaps could/should also rename this to LinkMessageParser?

Comment (michael-loki) [reply llarp/messages/link_message_parser.cpp]: Will do.

**Comment by tewinget** [comment]
for the files which were moved from `messages/` consider adding `_messages` to those which describe messages.

**Comment by majestrate** [comment]
ready?

**Comment by michael-loki** [comment]
yarp

**Pull request merged**